Mathlib/AlgebraicGeometry/EllipticCurve/VariableChange.lean

```diff
@@ -64,169 +64,195 @@ namespace VariableChange
 
 variable (C C' : VariableChange R)
 
+instance : One (VariableChange R) where
+  one := ⟨1, 0, 0, 0⟩
+
 /-- The identity linear change of variables given by the identity matrix. -/
-def id : VariableChange R :=
-  ⟨1, 0, 0, 0⟩
+lemma one_def : (1 : VariableChange R) = ⟨1, 0, 0, 0⟩ := rfl
+
+instance : Mul (VariableChange R) where
+  mul C C' := {
+    u := C.u * C'.u
+    r := C.r * C'.u ^ 2 + C'.r
+    s := C'.u * C.s + C'.s
+    t := C.t * C'.u ^ 3 + C.r * C'.s * C'.u ^ 2 + C'.t }
 
 /-- The composition of two linear changes of variables given by matrix multiplication. -/
-def comp : VariableChange R where
-  u := C.u * C'.u
-  r := C.r * C'.u ^ 2 + C'.r
-  s := C'.u * C.s + C'.s
-  t := C.t * C'.u ^ 3 + C.r * C'.s * C'.u ^ 2 + C'.t
+lemma mul_def : C * C' = {
```

Comment:
Is this (and `inv_def` and `variableChange_def`) a bit clunky? If you have `mul/inv_u/r/s/t` etc (just like `variableChange_a1/2/3/4/6`) then maybe you won't ever need this?

Reply:
*ping* any more feedbacks?

Reply:
Oops - no more feedback! It LGTM already :) but unfortunately I don't have the power to maintainer merge...